models/surveys.go

```diff
@@ -14,13 +36,88 @@ type Survey struct {
 	Reference string `json:"surveyRef"`
 }
 
-// AllSurveys returns summaries of all known surveys. The surveys are returned in ascending short name order.
-func AllSurveys() ([]*SurveySummary, error) {
-	rows, err := db.Query("SELECT id, shortname FROM survey.survey ORDER BY shortname ASC")
+type API struct {
+	AllSurveysStmt                    *sql.Stmt
+	GetSurveyStmt                     *sql.Stmt
+	GetSurveyByShortNameStmt          *sql.Stmt
+	GetSurveyByReferenceStmt          *sql.Stmt
+	GetSurveyIDStmt                   *sql.Stmt
+	GetClassifierTypeSelectorStmt     *sql.Stmt
+	GetClassifierTypeSelectorByIdStmt *sql.Stmt
+}
+
+func NewAPI(db *sql.DB) (*API, error) {
+	allSurveysSQL := "SELECT id, shortname FROM survey.survey ORDER BY shortname ASC"
+	allSurveyStmt, err := createStmt(allSurveysSQL, db)
+	if err != nil {
+		return nil, err
+	}
+
+	getSurveySQL := "SELECT id, shortname, longname, surveyref from survey.survey WHERE id = $1"
+	getSurveyStmt, err := createStmt(getSurveySQL, db)
+	if err != nil {
+		return nil, err
+	}
+
+	getSurveyByShortNameSQL := "SELECT id, shortname, longname, surveyref from survey.survey WHERE LOWER(shortName) = LOWER($1)"
+	getSurveyByShortNameStmt, err := createStmt(getSurveyByShortNameSQL, db)
+	if err != nil {
+		return nil, err
+	}
+
+	getSurveyByReferenceSQL := "SELECT id, shortname, longname, surveyref from survey.survey WHERE LOWER(surveyref) = LOWER($1)"
+	getSurveyByReferenceStmt, err := createStmt(getSurveyByReferenceSQL, db)
+	if err != nil {
+		return nil, err
+	}
+
+	getSurveyIdSQL := "SELECT id FROM survey.survey WHERE id = $1"
```

Comment:
If these strings aren't used again it's not worth assigning them - just run them inline in the `createStmt` calls